crates/core/src/contracts/erc20/ethereum_erc20.rs

```diff
@@ -0,0 +1,42 @@
+use ethers::utils::id;
+use reth_primitives::{Bytes, U256};
+use starknet::core::types::BlockId;
+use starknet::providers::Provider;
+use starknet_crypto::FieldElement;
+
+use crate::client::errors::EthApiError;
+use crate::client::helpers::DataDecodingError;
+use crate::contracts::kakarot::KakarotContract;
+use crate::models::felt::Felt252Wrapper;
+
+/// Abstraction for a Kakarot ERC20 contract.
+pub struct EthereumErc20<'a, P> {
+    pub address: FieldElement,
+    kakarot_contract: &'a KakarotContract<P>,
+}
+
+impl<'a, P: Provider + Send + Sync> EthereumErc20<'a, P> {
```

Comment:
isn't there any contract utils in `ethers` so we don't need to do any encoding ourself?

Reply:
I found [this](https://docs.rs/ethers/latest/ethers/abi/struct.Function.html#method.encode_input) which could help, but it means we need to have the abi.json.

Reply:
I would not be shocked to add an `IERC20.json` asset, or a rust version of it, at the end it's what we do manually, and it may be cleaner like "one the one hand this is the interface, one the other hand these are rust object to play with it"